src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -324,6 +342,14 @@ protected override Task WriteResponseHeadersAsyncCore(Metadata responseHeaders)
 
         public void Initialize()
         {
+            var currentActivity = Activity.Current;
+            if (currentActivity != null)
+            {
+                Activity.Current.AddTag("GrpcMethod", MethodCore);
```

Comment:
similar to the above 

1. Do we know this is the right activity? 
2. `grpc.method`

Reply:
Kestrel will start an activity, but there is the potential for someone else to start a different activity in middleware, which this logic would add a tag to. Is that a problem? Should this code check what the current activities name is? Instead of appending tags, should gRPC on the server always start a new activity just to be safe?

Reply:
It's possible to traverse back to the first Activity that does not have Parent and it will be the one we care about. In most cases, it will be the Current one and won't create any overhead.

Creating new Activity means we'll need new DiagnosticsSource events, tracing tools will need to do some work to onboard, and it seems to create a lot of work and overhead without providing visible value. 
I guess if we'll need to host gRPC server without AspNetCore Hosting, new Activities it will be needed